# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasource/TrackBuffersManager.h
# Commit: fd6a6b881725
# Full Hash: fd6a6b8817251844e298287d5baab81b23c9f838
# Author: Bryce Seager van Dyk <bvandyk@mozilla.com>
# Date: 2019-09-20 03:57:39
# Description:
#   Bug 1578143 - Prevent TrackBuffersManager attempting to access buffered ranges if none exist when evicting data. r=jya
#   
#   It appears possible for data eviction to take place despite the manager not
#   currently having any buffered ranges. This shouldn't happen, but if it can if
#   the buffers have become full with 0 duration data. This patch attempts to
# ==============================================================================

diff -r 708c110cf4ea -r fd6a6b881725 dom/media/mediasource/TrackBuffersManager.h
--- a/dom/media/mediasource/TrackBuffersManager.h	Thu Sep 19 16:39:50 2019 +0000
+++ b/dom/media/mediasource/TrackBuffersManager.h	Thu Sep 19 15:38:40 2019 +0000
@@ -194,6 +194,9 @@
   RefPtr<RangeRemovalPromise> CodedFrameRemovalWithPromise(
       media::TimeInterval aInterval);
   bool CodedFrameRemoval(media::TimeInterval aInterval);
+  // Removes all coded frames -- this is not to spec and should be used as a
+  // last resort to clear buffers only if other methods cannot.
+  void RemoveAllCodedFrames();
   void SetAppendState(SourceBufferAttributes::AppendState aAppendState);
 
   bool HasVideo() const { return mVideoTracks.mNumTracks > 0; }
